packages/shared/src/hooks/post/useViewPost.ts

```diff
@@ -3,40 +3,40 @@ import {
   useMutation,
   useQueryClient,
 } from '@tanstack/react-query';
-import { sendViewPost } from '../../graphql/posts';
+import { Post, sendViewPost } from '../../graphql/posts';
 import { generateQueryKey, RequestKey } from '../../lib/query';
 import { useAuthContext } from '../../contexts/AuthContext';
 import { ReadingDay, UserStreak } from '../../graphql/users';
 import { getTodayTz } from '../../lib/dateFormat';
 
-export const useViewPost = (): UseMutateAsyncFunction<
-  unknown,
-  unknown,
-  string
-> => {
+export const useViewPost = (
+  post: Post,
+): UseMutateAsyncFunction<unknown, unknown, string> => {
   const client = useQueryClient();
   const { user } = useAuthContext();
   const streakKey = generateQueryKey(RequestKey.UserStreak, user);
   const readKey = generateQueryKey(RequestKey.ReadingStreak30Days, user);
   const { mutateAsync: onSendViewPost } = useMutation(sendViewPost, {
     onSuccess: async () => {
       await client.invalidateQueries(streakKey);
-      client.setQueryData<UserStreak>(streakKey, (data) => {
-        const streak = { ...data };
-
-        if (
-          !streak?.lastViewAt ||
-          new Date(streak.lastViewAt).getDate() !== new Date().getDate()
-        ) {
-          streak.max += 1;
-          streak.total += 1;
-          streak.current += 1;
-        }
-
-        streak.lastViewAt = new Date();
-
-        return streak;
-      });
+      if (!post.read) {
```

Comment:
Just checking that this does not become read once user clicks on the post to read, thus disabling the cache to update in any case (if the post is being read now, or was read before)

Reply:
I tested this manually and it works as expected. E.g. if the post was not read before, it will update the cache correctly.